engine/runtime/src/main/java/org/enso/interpreter/runtime/ResourceManager.java

```diff
@@ -177,29 +177,34 @@ private final class ProcessItems extends ThreadLocalAction implements Runnable {
      */
     @Override
     protected void perform(ThreadLocalAction.Access access) {
+      var isMyThreadChoosen = false;
       for (; ; ) {
         Item[] toProcess;
         synchronized (pendingItems) {
-          request.cancel(false);
+          if (!isMyThreadChoosen) {
+            if (request == null || request.isCancelled() || request.isDone()) {
+              // some thread is already handing the request
```

Comment:
To continue the analysis at [previous PR](https://github.com/enso-org/enso/pull/10602/files#r1690926438):

> For recurring actions ... [Future.isDone()](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/concurrent/Future.html#isDone()) will return true only if the action was canceled.

Calling `request.cancel(false)` in the `synchronized (pendingItems)` seciton and then checking if it `isDone()` or `isCancelled()` when other threads arrive.